### PR TITLE
[4.3] php-diff line endings

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #40065 .

### Summary of Changes
update php-diff from @jfcherng to force the diff to ignore line ending differences

### Testing Instructions
@RickR2H please see if this solves your problem. I can't test it myself as I can't replicate the reported problem. But the logic of the change makles sense

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on aa78dca6c0bbf0883f67b9997ffc23f7f29fee69<br /><br />Can be tested by changing the line ending of the override file from LF to CRLF and vise versa. @ricardo1709<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

File: administrator/components/com_templates/tmpl/template/default.php [hunk 1]
```diff
@@ -161,6 +164,7 @@
                                         'detailLevel' => 'word',
                                         'spaceToHtmlTag' => true,
                                         'wrapperClasses' => ['diff-wrapper', 'columns-order-ignore'],
+
```
Comment (jfcherng) [inline administrator/components/com_templates/tmpl/template/default.php]: mysterious empty line :wink:

**Comment by ricardo1709** [comment]
I have tested this item :white_check_mark: successfully on aa78dca6c0bbf0883f67b9997ffc23f7f29fee69<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

**Comment by RickR2H** [comment]
@brianteeman please remove the empty line. I'll reset the tests as it is only code style change.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

**Comment by RickR2H** [comment]
Restored tests because change was just an empty line removal.
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

**Comment by obuisard** [comment]
There is an issue with the composer lock file. Can you take a look, Brian @brianteeman? Thanks!

**Comment by jfcherng** [comment]
Fyi, php-diff just release `6.10.10`. It shifts `ignoreLineEnding`'s implementation to its dependency.

**Comment by brianteeman** [comment]
@jfcherng just to confirm. If I update to 6.10.10 there is nothing else I need to change?

**Comment by jfcherng** [comment]
> @jfcherng just to confirm. If I update to 6.10.10 there is nothing else I need to change?

No. It just works.

**Comment by brianteeman** [comment]
thanks for confirming. I will update this when I get back to my computer

**Comment by obuisard** [comment]
@RickR2H can you test again? Thanks!

**Comment by obuisard** [comment]
I have tested this item :white_check_mark: successfully on 64f2296de4979b9270fb20bb5d3507e32848a0ec<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

**Comment by jfcherng** [comment]
> Jack @jfcherng do you mind testing this update? Thanks!

I don't use Joomla. I am just the maintainer of jfcherng/php-diff library. 😄

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 64f2296de4979b9270fb20bb5d3507e32848a0ec<br /><br />Works as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

**Comment by RickR2H** [comment]
@obuisard The RTC label was not removed with the latest PR. But with two successful tests it's again RTC.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40070">issues.joomla.org/tracker/joomla-cms/40070</a>.</sub>

**Comment by obuisard** [comment]
Thank you Brian @brianteeman

**Comment by brianteeman** [comment]
thanks @jfcherng for the update

**Pull request merged**